core/service/src/config.rs

```diff
@@ -170,6 +176,27 @@ impl<C, G, E> Configuration<C, G, E> where
 	pub fn client_id(&self) -> String {
 		format!("{}/v{}", self.impl_name, self.full_version())
 	}
+
+	/// Generate a PathBuf to sub in the chain configuration directory
+	/// if given
+	pub fn in_chain_config_dir(&self, sub: &str) -> Option<PathBuf> {
+		self.config_dir.clone().map(|mut path| {
+			path.push("chains");
+			path.push(self.chain_spec.id());
+			path.push(sub);
+			path
+		})
+	}
+	
+	/// The basepath for the db directory
+	pub fn database_path(&self) -> Option<PathBuf> {
```

Comment:
This function really doesn't make sense. There's already a `database.path` field in `self`, and what I would expect by  calling this function is that the value of this field gets returned.

In my opinion we should remove `path` from `DatabaseConfig` and split `NetworkConfiguration` in two, with one version without the paths in `Configuration` and one with the paths passed when building the network. This way, there would be on incoherent state in `Configuration`.

As a quick hack I'd be fine with `NetworkConfiguration::config_path`/`NetworkConfiguration::net_config_path` set to null, then appropriately set later.

Reply:
I am fine with removing that function and have the setter function directly call `in_chain_config_dir("db")` analog to how this is done for `keystore_path`. It was just meant as a handy function to have the `db` only be written once rather than multiple places throughout the code base.